src/manipulators/no_implicit_any_manipulator.ts

```diff
@@ -26,9 +37,286 @@ export class NoImplicitAnyManipulator extends Manipulator {
+          );
+        }
+      });
+    }
+  }
+
+  /**
+   * Adds a new dependency (edge) between two declarations.
+   * @param {Map<AcceptedDeclaration, Set<AcceptedDeclaration>>} dependencyGraph - Dependency graph.
+   * @param {Map<AcceptedDeclaration, Set<Type>>} determinedTypes - Calculated types for each declaration.
+   * @param {Set<AcceptedDeclaration>} modifiedDeclarations - Set of modified declarations (vertices).
+   * @param {Node<ts.Node>} predecessor - Declaration that successor is dependent on.
+   * @param {AcceptedDeclaration} successor - Declaration that is dependent on successor.
+   */
+  private addDependency(
+    dependencyGraph: Map<AcceptedDeclaration, Set<AcceptedDeclaration>>,
+    determinedTypes: Map<AcceptedDeclaration, Set<Type>>,
+    modifiedDeclarations: Set<AcceptedDeclaration>,
+    predecessor: Node<ts.Node>,
+    successor: AcceptedDeclaration
+  ): void {
+    // Get predecessor's type.
+    const determinedType = this.toTypeList(predecessor.getType());
+
+    // If predecessor is type any, add predecessor's declaration to the dependency graph and modified declarations set
+    // and add the dependency into the graph.
+    if (determinedType.some(type => type.isAny())) {
+      if (Node.isIdentifier(predecessor)) {
+        predecessor
+          .getSymbol()
+          ?.getDeclarations()
+          ?.forEach(assignedDeclaration => {
+            if (
+              Node.isVariableDeclaration(assignedDeclaration) ||
+              Node.isParameterDeclaration(assignedDeclaration)
+            ) {
+              modifiedDeclarations.add(assignedDeclaration);
+              this.addToMapSet(dependencyGraph, assignedDeclaration, successor);
+            }
+          });
+      }
+    } else {
+      // Otherwise, only add predecessor's type to successor's determined types.
+      determinedType.forEach(type => {
+        this.addToMapSet(determinedTypes, successor, type);
+      });
+    }
+  }
+
+  /**
+   * Topologically sorts a directed graph. If cyclic, returns subset of vertices not a part of a cycle.
+   * @param {Set<T>} vertices - Set of vertices.
+   * @param {Map<T, Set<T>>} edges - Map of directed edges between vertices.
+   * @return {T[]} Topologically sorted list of vertices.
+   */
+  topoSort<T>(vertices: Set<T>, edges: Map<T, Set<T>>): T[] {
+    const sorted: T[] = [];
+    const numDependencies = new Map<T, number>();
+
+    // Construct a map of in-degrees for each vertex.
+    for (const vertex of vertices) {
+      numDependencies.set(vertex, 0);
+    }
+
+    for (const successors of edges.values()) {
+      successors.forEach(successor => {
+        numDependencies.set(
+          successor,
+          (numDependencies.get(successor) || 0) + 1
+        );
+      });
+    }
+
+    // Add all vertices with in-degree to queue.
+    const queue: T[] = [];
+    for (const vertex of numDependencies.keys()) {
+      if (numDependencies.get(vertex) === 0) {
```

Comment:
suggestion: can be a filter

Reply:
I ended up reimplementing topoSort to handle cyclical dependencies better, but thanks for the suggestion!